### ISSUE TITLE
Do Pre-training XLNet in TF1.4 and PY3 But get Loss Nan

### ISSUE THREAD

**Issue by ft3020997** [zihangdai/xlnet#170]
Hi,
 I'm doing pretraining with Chinese Corpus on 4 v100 32G GPU. All of steps are followed the guide command in ReadMe, including train sentencePieces, use data_util to preprocess trainData and run train_gpu.py. The params are listed as below:
![image](https://user-images.githubusercontent.com/21030532/61423684-44948d80-a943-11e9-890d-4f38d1f7829f.png)

But, i get the NAN loss through training steps, as below pic :
![image](https://user-images.githubusercontent.com/21030532/61423486-8d981200-a942-11e9-9d2d-b7f87817b415.png)
 is there any mistake happened? 

Thx for any help.

**Comment by dhlee347** [comment]
try it with TF 1.13

**Comment by Stodgers** [comment]
我在tf1.14下tpu模块引用出错，换成1.13之后报错TypeError: Expected binary or unicode string, got None。
老哥能不能提供一份预训练的代码，参考一下。

**Comment by MissMcFly** [comment]
请问预训练的python版本是2.7，tensorflow是1.13吗？谢谢~

**Comment by ft3020997** [comment]
> 我在tf1.14下tpu模块引用出错，换成1.13之后报错TypeError: Expected binary or unicode string, got None。
> 老哥能不能提供一份预训练的代码，参考一下。

我也是使用T1.4但是没有遇到你说的这个问题，预训练的代码跟官方代码一致，主要是需要自己训练sentencePieces，这个比较关键。不过我现在这个问题已经解决了，是训练集的问题。

**Comment by ft3020997** [comment]
> 请问预训练的python版本是2.7，tensorflow是1.13吗？谢谢~

我用的是PY3.0和TF1.4版本，目前这问题已经解决了。

**Comment by 008karan** [comment]
任何人都可以告诉他们如何训练Sentencepeice模型？

**Comment by Stodgers** [comment]
> > 我在tf1.14下tpu模块引用出错，换成1.13之后报错TypeError: Expected binary or unicode string, got None。
> > 老哥能不能提供一份预训练的代码，参考一下。
> 
> 我也是使用T1.4但是没有遇到你说的这个问题，预训练的代码跟官方代码一致，主要是需要自己训练sentencePieces，这个比较关键。不过我现在这个问题已经解决了，是训练集的问题。

能不能私聊一波，sentencePieces训练那边有很多问题，就比如训练前要不要分词，然后语料需不需要按照段落中间空行等等。我Q 1879543220 跪谢orz

**Comment by Ina299** [comment]
I had the same problem on training Japanese corpus.
I recommend you to train your sentence piece model on your own corpus. ( I solved Nan problem by doing so.)
And training sentence piece does not take time as I expected.
